techlibs/xilinx/synth_xilinx.cc

```diff
@@ -260,32 +274,65 @@ struct SynthXilinxPass : public ScriptPass
 			run("memory_map");
 			run("dffsr2dff");
 			run("dff2dffe");
+			if (help_mode) {
+				run("simplemap t:$mux", "         ('-widemux' only)");
+				run("muxcover <internal options>, ('-widemux' only)");
+			}
+			else if (widemux > 0) {
+				run("simplemap t:$mux");
+				std::string muxcover_args = " -nodecode";
+				switch (widemux) {
+					// NB: Cost of mux2 is 100; mux8 should cost between 3 and 4
+					//     of those so that 4:1 muxes and below are implemented
+					//     out of mux2s
+					case  5: muxcover_args += " -mux8=350 -mux16=400"; break;
```

Comment:
Wouldn't it be neater to calculate these rather than have a hard-to-maintain enumeration of possibilities here?